src/briefcase/commands/build.py

```diff
@@ -14,16 +17,66 @@ def add_options(self, parser):
         self._add_update_options(parser, context_label=" before building")
         self._add_test_options(parser, context_label="Build")
 
-    def build_app(self, app: BaseConfig, **options):
+    def build_app(self, app: AppConfig, **options):
         """Build an application.
 
         :param app: The application to build
         """
         # Default implementation; nothing to build.
 
+    def verify_template(self, app: AppConfig):
+        """Verify the template satisfies the Command's requirements."""
+        self.logger.info("Checking template compatability...", prefix=app.app_name)
+
+        with self.input.wait_bar("Verifying Briefcase support..."):
```

Comment:
Not sure a waitbar is needed here. Verification should be near immediate.

Reply:
IMO, this wait bar here is mostly intended to provide better continuity with the rest of the output.

While I consider the Wait Bar to be mostly about communicating a task is ongoing, I think the consistency of `[task description]... done` helps users understand what exactly is taking place in a consistent format. There's several small steps like this we're already wrapping with the Wait Bar elsewhere.

I suppose the end state of the Wait bar output could be manually printed...but that feels a little awkward.

Still want to remove/change this?

Example output:
```
[helloworld] Generating application template...
Using app template: /home/russell/github/beeware/briefcase-android-gradle-template, branch v0.3.15

[helloworld] Installing support package...
No support package required.

[helloworld] Installing application code...
Installing src/helloworld... done

[helloworld] Installing requirements...
Writing requirements file... done

[helloworld] Installing application resources...
Unable to find src/helloworld/resources/helloworld-round-48.png for 48px round application icon; using default
Unable to find src/helloworld/resources/helloworld-round-72.png for 72px round application icon; using default
Unable to find src/helloworld/resources/helloworld-round-96.png for 96px round application icon; using default
Unable to find src/helloworld/resources/helloworld-round-144.png for 144px round application icon; using default
Unable to find src/helloworld/resources/helloworld-round-192.png for 192px round application icon; using default
Unable to find src/helloworld/resources/helloworld-square-48.png for 48px square application icon; using default
Unable to find src/helloworld/resources/helloworld-square-72.png for 72px square application icon; using default
Unable to find src/helloworld/resources/helloworld-square-96.png for 96px square application icon; using default
Unable to find src/helloworld/resources/helloworld-square-144.png for 144px square application icon; using default
Unable to find src/helloworld/resources/helloworld-square-192.png for 192px square application icon; using default

[helloworld] Removing unneeded app content...
Removing directory app/src/main/python/helloworld/__pycache__
Removing unneeded app bundle content... done

[helloworld] Created build/helloworld/android/gradle

[helloworld] Checking template compatibility...
Verifying Briefcase support... done

[helloworld] Updating app metadata...
Setting main module... done

[helloworld] Building Android APK...
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details
```

Reply:
I see what you're saying; I guess for me, "is this a valid template" is a fast validation check that should almost always pass. It's not "doing" anything - it's just verifying that everything else will work. 99% of the time, it won't be communicating anything - and the 1% of the time that it does, there will be a big "it won't work" error message. 

Some of the other "short lived" tasks (e.g., updating app metadata) are actually *doing* something - they're part of the build process, writing/modifiying files, etc. I can't think of another task that is (a) short lived and (b) doesn't write/add/delete anything to the disk, with the exception of no-op tasks that are flagged has doing something, but then the platform backend doesn't need to actually do anything. 

The closest I can think of is some of the Docker checks - but those (a) *can* take time, and (b) can fail in some creative ways, so flagging that you're in the middle of a Docker check can be useful debugging.

Reply:
Ok...so, do you want this to print anything then? I feel like just printing the section header `[helloworld] Checking template compatibility...` with nothing else is also awkward.

That makes sense if so.

Reply:
Oh sorry - I thought that part was implied. My intention was to say that there should be *no* message when verifying the template version, unless there's a problem. Apologies for the ambiguity on my part.